src/DataGrid.tsx

```diff
@@ -502,11 +519,17 @@ function DataGrid<R, SR, K extends Key>(
       style.gridTemplateColumns = newTemplateColumns.join(' ');
     }
 
-    if (columnWidths.get(column.key) === measuredWidth) return;
+    if (resizedColumnWidths.get(column.key) === measuredWidth) return;
 
-    const newColumnWidths = new Map(columnWidths);
-    newColumnWidths.set(column.key, measuredWidth);
-    setColumnWidths(newColumnWidths);
+    const newResizedColumnWidths = new Map(resizedColumnWidths);
+    newResizedColumnWidths.set(column.key, measuredWidth);
+    setResizedColumnWidths(newResizedColumnWidths);
+
+    // We need to reset measured width otherwise grid used the already measured
+    // width in the next render and the new measured width is never set
+    if (resetMeasuredColumnWidths) {
+      setMeasuredColumnWidths(new Map());
```

Comment:
Rather than setting an empty map, can we immediately measure the new widths here?
Might save us an extra re-render